pkg/extract/nvd/api/cve/cve.go

```diff
@@ -41,24 +87,350 @@ func Extract(args string, opts ...Option) error {
 		return errors.Wrapf(err, "remove %s", options.dir)
 	}
 
-	log.Printf("[INFO] Extract ")
-	if err := filepath.WalkDir(args, func(path string, d fs.DirEntry, err error) error {
-		if err != nil {
-			return err
-		}
+	log.Printf("[INFO] Extract NVD API CVE")
+
+	g, ctx := errgroup.WithContext(context.Background())
+	g.SetLimit(options.concurrency)
+
+	reqChan := make(chan string)
+	g.Go(func() error {
+		defer close(reqChan)
+		if err := filepath.WalkDir(cveDir, func(path string, d fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
 
-		if d.IsDir() {
+			if d.IsDir() || filepath.Ext(path) != ".json" {
+				return nil
+			}
+
+			select {
+			case reqChan <- path:
+			case <-ctx.Done():
+				return ctx.Err()
+			}
 			return nil
+		}); err != nil {
+			return errors.Wrapf(err, "walk %s", cveDir)
 		}
+		return nil
+	})
+
+	for i := 0; i < options.concurrency; i++ {
+		g.Go(func() error {
+			for path := range reqChan {
+				f, err := os.Open(path)
+				if err != nil {
+					return errors.Wrapf(err, "open %s", path)
+				}
+				defer f.Close()
+
+				var fetched cveTypes.CVE
+				if err := json.NewDecoder(f).Decode(&fetched); err != nil {
+					return errors.Wrapf(err, "decode %s", path)
+				}
 
-		if filepath.Ext(path) != ".json" {
+				data, err := extract(fetched, cpematchDir)
+				if err != nil {
+					return errors.Wrapf(err, "extract %s", path)
+				}
+
+				ss := strings.Split(data.ID, "-")
+				if len(ss) < 3 || ss[0] != "CVE" {
+					return errors.Errorf("unexpected ID format. expected: %q, actual: %q", "CVE-<year>-<ID>", data.ID)
+				}
+
+				if err := util.Write(filepath.Join(options.dir, "data", ss[1], fmt.Sprintf("%s.json", data.ID)), data, true); err != nil {
+					return errors.Wrapf(err, "write %s", filepath.Join(options.dir, "data", ss[1], fmt.Sprintf("%s.json", data.ID)))
+				}
```

Comment:
```suggestion
				if err := util.Write(filepath.Join(options.dir, "data", filepath.Base(filepath.Dir(path)), fmt.Sprintf("%s.json", data.ID)), data, true); err != nil {
					return errors.Wrapf(err, "write %s", filepath.Join(options.dir, "data", filepath.Base(filepath.Dir(path)), fmt.Sprintf("%s.json", data.ID)))
				}
```

Reply:
done in d7aa76d (but it will disappear by rebase)